ang/volunteer/Project.js

```diff
@@ -91,20 +82,29 @@
       if (CRM.vars['org.civicrm.volunteer'].entityTitle) {
         project.title = CRM.vars['org.civicrm.volunteer'].entityTitle;
       }
-      //VOL-223
-      $scope.editContactType = supporting_data.values.canEditContacts;
     } else {
-      //VOL-223
-      $scope.editContactType = can_edit_contact_types.values;
       $(relationship_data.values).each(function (index, relationship) {
-        if (!relationships.hasOwnProperty(relationship.relationship_type_id)) {
-          relationships[relationship.relationship_type_id] = [];
+        if (!volRelData.hasOwnProperty(relationship.relationship_type_id)) {
+          volRelData[relationship.relationship_type_id] = [];
         }
-        relationships[relationship.relationship_type_id].push(relationship.contact_id);
+        volRelData[relationship.relationship_type_id].push({
+          contact_id: relationship.contact_id,
+          can_be_read_by_current_user: relationship.can_be_read_by_current_user
+        });
       });
-      var originalRelationships = _.clone(relationships);
     }
+
+    // flatten the data a bit to make it easier to work with in the template
+    _.each(volRelData, function (contacts, relTypeId) {
+      relationships[relTypeId] = [];
+      showRelationshipType[relTypeId] = false;
+      _.each(contacts, function (contact) {
+        relationships[relTypeId].push(contact.contact_id);
+        showRelationshipType[relTypeId] = showRelationshipType[relTypeId] || contact.can_be_read_by_current_user;
```

Comment:
Took me a couple reads to get this. Nice catch -- I wasn't aware that I changed the logic here.

I think you're right that changing this to an OR could have undesirable consequences. Could you merge this into your repo and commit a fix (thus updating the PR into the main repo)? Looks like you'll need to set line 100 to `true` as well.